br/pkg/lightning/restore/check_info.go

```diff
@@ -399,535 +180,64 @@ func (rc *Controller) estimateSourceData(ctx context.Context) (int64, int64, boo
 }
 
 // localResource checks the local node has enough resources for this import when local backend enabled;
-func (rc *Controller) localResource(ctx context.Context, sourceSize int64) error {
```

Comment:
The logic is separated into two checkers.  The first check is `(*localDiskPlacementCheckItem).Check()`, in `br/pkg/lightning/restore/precheck_impl.go`, L443; The second check is `(*localTempKVDirCheckItem).Check()`, in `br/pkg/lightning/restore/precheck_impl.go`, L479